Greatly reduce GCC build logs

GuillaumeGomez opened the pull request:
Fixes https://github.com/rust-lang/rust-log-analyzer/issues/80.

Based on [makefile documentation](https://www.gnu.org/software/make/manual/html_node/Options-Summary.html#index-_002d_002dquiet-1) and [configure documentation](https://www.gnu.org/software/autoconf/manual/autoconf-2.68/html_node/configure-Invocation.html).

cc @RalfJung @antoyo

GuillaumeGomez committed:
Greatly reduce GCC build logs

RalfJung commented on src/ci/docker/host-x86_64/dist-x86_64-linux/build-gccjit.sh:
```diff
@@ -21,8 +21,9 @@ cd ../gcc-build
     --disable-bootstrap \
     --disable-multilib \
     --prefix=$(pwd)/../gcc-install
```
Doesn't this need a backslash? Otherwise you're calling a new command `--quiet`...

GuillaumeGomez replied:
Arf, thanks!

GuillaumeGomez commented:
The `--quiet` option isn't enough apparently... Let's do the good old `> /dev/null`.

RalfJung commented:
That will be annoying if it ever fails though...
Is there an upstream issue tracking "please provide a build option that outputs less than 100 lines on success"?

antoyo commented:
> That will be annoying if it ever fails though... Is there an upstream issue tracking "please provide a build option that outputs less than 100 lines on success"?

The errors are printed on stderr, so it's not an issue imo.

RalfJung commented:
In my experience the output often becomes hard to follow when throwing away just stdout, as nobody tests that.

But sure, we can give it a shot.

RalfJung commented on src/ci/docker/host-x86_64/dist-x86_64-linux/build-gccjit.sh:
```diff
@@ -20,9 +20,10 @@ cd ../gcc-build
+    --prefix=$(pwd)/../gcc-install > /dev/null
+
+make > /dev/null
+make install > /dev/null
```
Are *all* of these steps generating thousands of lines?

Also please add a comment explaining why we're taking such drastic measures.

antoyo replied:
The configure (if we added back `--quiet`) might give only a dozen lines, but `make` will definitely generate thousands of lines (even with `--quiet`). For `make install`, it might be only a couple hundred lines with `--quiet`.

GuillaumeGomez replied:
I still think it's way too much (for all commands). So better keep it this way as it's not relevant to the build.

I'll add comments.

RalfJung replied:
> The configure (if we added back --quiet) might give only a dozen lines,

Seems reasonable to just keep that visible then.

GuillaumeGomez replied:
Ok, done.

antoyo replied:
You forgot to add back `--quiet` to configure, no?

GuillaumeGomez replied:
Not sure it changes anything but let's put it back.

TimNN commented:
I don't know if it's easily installable in whatever OS the CI Docker container uses, but there's a tool called `chronic` that

> runs a command, and arranges for its standard out and standard error to only be
>       displayed if the command fails (exits nonzero or crashes).  If the command succeeds, any
 >     extraneous output will be hidden.

Which seems like it would be useful here.

https://manpages.ubuntu.com/manpages/noble/man1/chronic.1.html

klensy commented:
Well, you can tune log analyzer to skip these build logs instead turning off logs.

RalfJung commented:
Sometimes one has to look at the logs though. I don't think it is reasonable to have 80% of them filled with the GCC build when that's a tiny part of what we are testing.

klensy commented:
It wasn't an issue before, so there something changed in pipeline?

RalfJung commented:
It's not been that long since GCC was added to the build. I think it was a problem ever since, or do you have evidence to the contrary?

Mark-Simulacrum commented on src/ci/docker/host-x86_64/dist-x86_64-linux/build-gccjit.sh:
```diff
@@ -20,9 +20,14 @@ cd ../gcc-build
+# of output. Unfortunately, even when using `--quiet` option for all commands,
+# the output still thousands of lines of output, forcing us to use this solution.
+make > /dev/null
+make install > /dev/null
```
Please use hide_output from src/ci/docker/scripts/shared.sh (feel free to grep for other usages). That makes sure there's some progress indication and we do the right thing on failure.

RalfJung commented:
Btw is it expected that the GCC build takes about 30 minutes? (If I understand [this log](https://productionresultssa19.blob.core.windows.net/actions-results/949eab71-f9f1-4390-8a18-61665488acb3/workflow-job-run-a64ed217-e4bb-5680-9c18-8f3ddeed4bfc/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-03-15T06%3A44%3A26Z&sig=6iqHyHG2WZBjYeyDmihDDTH7WvdlYVXJEM1vzaW0BX4%3D&sp=r&spr=https&sr=b&st=2024-03-15T06%3A34%3A21Z&sv=2021-12-02) correctly.) That's 33% of the entire CI job! And in particular this happens *before* the rustc test suite is run, greatly increasing the latency until regular test failures are discovered.

RalfJung commented:
Also why do both the gnu-llvm-16 and the gnu-tools runner build GCC...?

GuillaumeGomez commented:
> Also why do both the gnu-llvm-16 and the gnu-tools runner build GCC...?

It is run it 3 CIs: gnu-tools, llvm-16 and llvm-17. When I made the last sync of `rustc_codegen_gcc`, I removed the installations of libgccjit through the system and replaced it with the build of GCC (you can see that in the last commits of #122042, like https://github.com/rust-lang/rust/pull/122042/commits/c5c6729c88b6d00798f46d5c7397e854775d0923).

If people agree on reducing the number of CIs building GCC, I can remove some of them.

GuillaumeGomez commented:
> Btw is it expected that the GCC build takes about 30 minutes? (If I understand [this log](https://productionresultssa19.blob.core.windows.net/actions-results/949eab71-f9f1-4390-8a18-61665488acb3/workflow-job-run-a64ed217-e4bb-5680-9c18-8f3ddeed4bfc/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-03-15T06%3A44%3A26Z&sig=6iqHyHG2WZBjYeyDmihDDTH7WvdlYVXJEM1vzaW0BX4%3D&sp=r&spr=https&sr=b&st=2024-03-15T06%3A34%3A21Z&sv=2021-12-02) correctly.) That's 30% of the entire CI job! And in particular this happens _before_ the rustc test suite is run, greatly increasing the latency until regular test failures are discovered.

For now it's built using only one thread. I don't know if I can get the number of CPUs available in the CI and pass it to the `make` scripts. That would likely greatly improve the time it takes in the CI.

GuillaumeGomez committed:
Build GCC with as many threads as available

rust-log-analyzer commented:
A job failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8294853377/job/22700665882) [(plain)](https://github.com/rust-lang/rust/commit/ede2082bb42dc208a043c05c3d179e7f16bbfa75/checks/22700665882/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#15 162.0 
#15 162.0 HEAD is now at 78dc50f0e50 Use full gith hash for release names
#15 162.0 + mkdir ../gcc-build ../gcc-install
#15 162.0 + cd ../gcc-build
#15 162.0 + . /tmp/shared.sh
#15 162.0 /scripts/build-gccjit.sh: 16: .: cannot open /tmp/shared.sh: No such file
#15 ERROR: process "/bin/sh -c sh /scripts/build-gccjit.sh /scripts" did not complete successfully: exit code: 2
 > [10/10] RUN sh /scripts/build-gccjit.sh /scripts:
162.0 
162.0   git switch -
162.0 
162.0 
162.0 Turn off this advice by setting config variable advice.detachedHead to false
162.0 
162.0 HEAD is now at 78dc50f0e50 Use full gith hash for release names
162.0 + mkdir ../gcc-build ../gcc-install
162.0 + cd ../gcc-build
162.0 + . /tmp/shared.sh
162.0 /scripts/build-gccjit.sh: 16: .: cannot open /tmp/shared.sh: No such file
Dockerfile:100
--------------------
  98 |     COPY host-x86_64/dist-x86_64-linux/build-gccjit.sh /scripts/
  99 |     
---
#14 159.4 
#14 159.4 HEAD is now at 78dc50f0e50 Use full gith hash for release names
#14 159.4 + mkdir ../gcc-build ../gcc-install
#14 159.4 + cd ../gcc-build
#14 159.4 + . /tmp/shared.sh
#14 159.4 /scripts/build-gccjit.sh: 16: .: cannot open /tmp/shared.sh: No such file
#14 ERROR: process "/bin/sh -c sh /scripts/build-gccjit.sh /scripts" did not complete successfully: exit code: 2
 > [10/10] RUN sh /scripts/build-gccjit.sh /scripts:
159.4 
159.4   git switch -
159.4 
159.4 
159.4 Turn off this advice by setting config variable advice.detachedHead to false
159.4 
159.4 HEAD is now at 78dc50f0e50 Use full gith hash for release names
159.4 + mkdir ../gcc-build ../gcc-install
159.4 + cd ../gcc-build
159.4 + . /tmp/shared.sh
159.4 /scripts/build-gccjit.sh: 16: .: cannot open /tmp/shared.sh: No such file
Dockerfile:100
--------------------
  98 |     COPY host-x86_64/dist-x86_64-linux/build-gccjit.sh /scripts/
  99 |     
---
#15 160.8 
#15 160.8 HEAD is now at 78dc50f0e50 Use full gith hash for release names
#15 160.8 + mkdir ../gcc-build ../gcc-install
#15 160.8 + cd ../gcc-build
#15 160.8 + . /tmp/shared.sh
#15 160.8 /scripts/build-gccjit.sh: 16: .: cannot open /tmp/shared.sh: No such file
#15 ERROR: process "/bin/sh -c sh /scripts/build-gccjit.sh /scripts" did not complete successfully: exit code: 2
 > [10/10] RUN sh /scripts/build-gccjit.sh /scripts:
160.8 
160.8   git switch -
160.8 
160.8 
160.8 Turn off this advice by setting config variable advice.detachedHead to false
160.8 
160.8 HEAD is now at 78dc50f0e50 Use full gith hash for release names
160.8 + mkdir ../gcc-build ../gcc-install
160.8 + cd ../gcc-build
160.8 + . /tmp/shared.sh
160.8 /scripts/build-gccjit.sh: 16: .: cannot open /tmp/shared.sh: No such file
Dockerfile:100
--------------------
  98 |     COPY host-x86_64/dist-x86_64-linux/build-gccjit.sh /scripts/
  99 |     
---
#14 157.5 
#14 157.5 HEAD is now at 78dc50f0e50 Use full gith hash for release names
#14 157.5 + mkdir ../gcc-build ../gcc-install
#14 157.5 + cd ../gcc-build
#14 157.5 + . /tmp/shared.sh
#14 157.5 /scripts/build-gccjit.sh: 16: .: cannot open /tmp/shared.sh: No such file
#14 ERROR: process "/bin/sh -c sh /scripts/build-gccjit.sh /scripts" did not complete successfully: exit code: 2
 > [10/10] RUN sh /scripts/build-gccjit.sh /scripts:
157.5 
157.5   git switch -
157.5 
157.5 
157.5 Turn off this advice by setting config variable advice.detachedHead to false
157.5 
157.5 HEAD is now at 78dc50f0e50 Use full gith hash for release names
157.5 + mkdir ../gcc-build ../gcc-install
157.5 + cd ../gcc-build
157.5 + . /tmp/shared.sh
157.5 /scripts/build-gccjit.sh: 16: .: cannot open /tmp/shared.sh: No such file
Dockerfile:100
--------------------
  98 |     COPY host-x86_64/dist-x86_64-linux/build-gccjit.sh /scripts/
  99 |     
---
#15 163.6 
#15 163.6 HEAD is now at 78dc50f0e50 Use full gith hash for release names
#15 163.6 + mkdir ../gcc-build ../gcc-install
#15 163.6 + cd ../gcc-build
#15 163.6 + . /tmp/shared.sh
#15 163.6 /scripts/build-gccjit.sh: 16: .: cannot open /tmp/shared.sh: No such file
#15 ERROR: process "/bin/sh -c sh /scripts/build-gccjit.sh /scripts" did not complete successfully: exit code: 2
 > [10/10] RUN sh /scripts/build-gccjit.sh /scripts:
163.6 
163.6   git switch -
163.6 
163.6 
163.6 Turn off this advice by setting config variable advice.detachedHead to false
163.6 
163.6 HEAD is now at 78dc50f0e50 Use full gith hash for release names
163.6 + mkdir ../gcc-build ../gcc-install
163.6 + cd ../gcc-build
163.6 + . /tmp/shared.sh
163.6 /scripts/build-gccjit.sh: 16: .: cannot open /tmp/shared.sh: No such file
Dockerfile:100
--------------------
  98 |     COPY host-x86_64/dist-x86_64-linux/build-gccjit.sh /scripts/
  99 |     
```

</details>

rust-log-analyzer commented:
A job failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8295125574/job/22701534593) [(plain)](https://github.com/rust-lang/rust/commit/cbe81e68fb0a7bb34828f0707bd5b3dbcd12fefa/checks/22701534593/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#16 156.5 + pwd
#16 156.5 + ../gcc-src/configure --enable-host-shared --enable-languages=jit --enable-checking=release --disable-bootstrap --disable-multilib --prefix=/scripts/gcc-build/../gcc-install --quiet
#16 157.1 The following languages will be built: c,jit,lto
#16 157.4 *** This configuration is not supported in the following subdirectories:
#16 157.4      gnattools gotools libgrust target-libada target-libstdc++-v3 target-libphobos target-zlib target-libbacktrace target-libgfortran target-libgo target-libffi target-libgm2 target-libobjc target-libgrust target-libitm target-libsanitizer target-libvtv
#16 157.5 /scripts/gcc-src/missing: 81: makeinfo: not found
#16 157.8 + hide_output make -j2
#16 157.8 + set +x
#16 157.8 + set +x
#16 157.8 trap: ERR: bad trap
#16 ERROR: process "/bin/sh -c sh /scripts/build-gccjit.sh /scripts" did not complete successfully: exit code: 1
 > [11/11] RUN sh /scripts/build-gccjit.sh /scripts:
156.5 + pwd
156.5 + ../gcc-src/configure --enable-host-shared --enable-languages=jit --enable-checking=release --disable-bootstrap --disable-multilib --prefix=/scripts/gcc-build/../gcc-install --quiet
157.1 The following languages will be built: c,jit,lto
157.1 The following languages will be built: c,jit,lto
157.4 *** This configuration is not supported in the following subdirectories:
157.4      gnattools gotools libgrust target-libada target-libstdc++-v3 target-libphobos target-zlib target-libbacktrace target-libgfortran target-libgo target-libffi target-libgm2 target-libobjc target-libgrust target-libitm target-libsanitizer target-libvtv
157.5 /scripts/gcc-src/missing: 81: makeinfo: not found
157.8 + hide_output make -j2
157.8 + set +x
157.8 + set +x
157.8 trap: ERR: bad trap
Dockerfile:101
--------------------
  99 |     COPY host-x86_64/dist-x86_64-linux/build-gccjit.sh /scripts/
 100 |     
---
#15 157.7 + pwd
#15 157.7 + ../gcc-src/configure --enable-host-shared --enable-languages=jit --enable-checking=release --disable-bootstrap --disable-multilib --prefix=/scripts/gcc-build/../gcc-install --quiet
#15 158.3 The following languages will be built: c,jit,lto
#15 158.7 *** This configuration is not supported in the following subdirectories:
#15 158.7      gnattools gotools libgrust target-libada target-libstdc++-v3 target-libphobos target-zlib target-libbacktrace target-libgfortran target-libgo target-libffi target-libgm2 target-libobjc target-libgrust target-libitm target-libsanitizer target-libvtv
#15 158.7 /scripts/gcc-src/missing: 81: makeinfo: not found
#15 159.1 + hide_output make -j2
#15 159.1 + set +x
#15 159.1 + set +x
#15 159.1 trap: ERR: bad trap
#15 ERROR: process "/bin/sh -c sh /scripts/build-gccjit.sh /scripts" did not complete successfully: exit code: 1
 > [11/11] RUN sh /scripts/build-gccjit.sh /scripts:
157.7 + pwd
157.7 + ../gcc-src/configure --enable-host-shared --enable-languages=jit --enable-checking=release --disable-bootstrap --disable-multilib --prefix=/scripts/gcc-build/../gcc-install --quiet
158.3 The following languages will be built: c,jit,lto
158.3 The following languages will be built: c,jit,lto
158.7 *** This configuration is not supported in the following subdirectories:
158.7      gnattools gotools libgrust target-libada target-libstdc++-v3 target-libphobos target-zlib target-libbacktrace target-libgfortran target-libgo target-libffi target-libgm2 target-libobjc target-libgrust target-libitm target-libsanitizer target-libvtv
158.7 /scripts/gcc-src/missing: 81: makeinfo: not found
159.1 + hide_output make -j2
159.1 + set +x
159.1 + set +x
159.1 trap: ERR: bad trap
Dockerfile:101
--------------------
  99 |     COPY host-x86_64/dist-x86_64-linux/build-gccjit.sh /scripts/
 100 |     
---
#16 156.8 + pwd
#16 156.8 + ../gcc-src/configure --enable-host-shared --enable-languages=jit --enable-checking=release --disable-bootstrap --disable-multilib --prefix=/scripts/gcc-build/../gcc-install --quiet
#16 157.3 The following languages will be built: c,jit,lto
#16 157.7 *** This configuration is not supported in the following subdirectories:
#16 157.7      gnattools gotools libgrust target-libada target-libstdc++-v3 target-libphobos target-zlib target-libbacktrace target-libgfortran target-libgo target-libffi target-libgm2 target-libobjc target-libgrust target-libitm target-libsanitizer target-libvtv
#16 157.7 /scripts/gcc-src/missing: 81: makeinfo: not found
#16 158.1 + hide_output make -j2
#16 158.1 + set +x
#16 158.1 + set +x
#16 158.1 trap: ERR: bad trap
#16 ERROR: process "/bin/sh -c sh /scripts/build-gccjit.sh /scripts" did not complete successfully: exit code: 1
 > [11/11] RUN sh /scripts/build-gccjit.sh /scripts:
156.8 + pwd
156.8 + ../gcc-src/configure --enable-host-shared --enable-languages=jit --enable-checking=release --disable-bootstrap --disable-multilib --prefix=/scripts/gcc-build/../gcc-install --quiet
157.3 The following languages will be built: c,jit,lto
157.3 The following languages will be built: c,jit,lto
157.7 *** This configuration is not supported in the following subdirectories:
157.7      gnattools gotools libgrust target-libada target-libstdc++-v3 target-libphobos target-zlib target-libbacktrace target-libgfortran target-libgo target-libffi target-libgm2 target-libobjc target-libgrust target-libitm target-libsanitizer target-libvtv
157.7 /scripts/gcc-src/missing: 81: makeinfo: not found
158.1 + hide_output make -j2
158.1 + set +x
158.1 + set +x
158.1 trap: ERR: bad trap
Dockerfile:101
--------------------
  99 |     COPY host-x86_64/dist-x86_64-linux/build-gccjit.sh /scripts/
 100 |     
---
#15 157.1 + pwd
#15 157.1 + ../gcc-src/configure --enable-host-shared --enable-languages=jit --enable-checking=release --disable-bootstrap --disable-multilib --prefix=/scripts/gcc-build/../gcc-install --quiet
#15 157.7 The following languages will be built: c,jit,lto
#15 158.0 *** This configuration is not supported in the following subdirectories:
#15 158.0      gnattools gotools libgrust target-libada target-libstdc++-v3 target-libphobos target-zlib target-libbacktrace target-libgfortran target-libgo target-libffi target-libgm2 target-libobjc target-libgrust target-libitm target-libsanitizer target-libvtv
#15 158.0 /scripts/gcc-src/missing: 81: makeinfo: not found
#15 158.4 + hide_output make -j2
#15 158.4 + set +x
#15 158.4 + set +x
#15 158.4 trap: ERR: bad trap
#15 ERROR: process "/bin/sh -c sh /scripts/build-gccjit.sh /scripts" did not complete successfully: exit code: 1
 > [11/11] RUN sh /scripts/build-gccjit.sh /scripts:
157.1 + pwd
157.1 + ../gcc-src/configure --enable-host-shared --enable-languages=jit --enable-checking=release --disable-bootstrap --disable-multilib --prefix=/scripts/gcc-build/../gcc-install --quiet
157.7 The following languages will be built: c,jit,lto
157.7 The following languages will be built: c,jit,lto
158.0 *** This configuration is not supported in the following subdirectories:
158.0      gnattools gotools libgrust target-libada target-libstdc++-v3 target-libphobos target-zlib target-libbacktrace target-libgfortran target-libgo target-libffi target-libgm2 target-libobjc target-libgrust target-libitm target-libsanitizer target-libvtv
158.0 /scripts/gcc-src/missing: 81: makeinfo: not found
158.4 + hide_output make -j2
158.4 + set +x
158.4 + set +x
158.4 trap: ERR: bad trap
Dockerfile:101
--------------------
  99 |     COPY host-x86_64/dist-x86_64-linux/build-gccjit.sh /scripts/
 100 |     
---
#16 157.6 + pwd
#16 157.6 + ../gcc-src/configure --enable-host-shared --enable-languages=jit --enable-checking=release --disable-bootstrap --disable-multilib --prefix=/scripts/gcc-build/../gcc-install --quiet
#16 158.2 The following languages will be built: c,jit,lto
#16 158.6 *** This configuration is not supported in the following subdirectories:
#16 158.6      gnattools gotools libgrust target-libada target-libstdc++-v3 target-libphobos target-zlib target-libbacktrace target-libgfortran target-libgo target-libffi target-libgm2 target-libobjc target-libgrust target-libitm target-libsanitizer target-libvtv
#16 158.6 /scripts/gcc-src/missing: 81: makeinfo: not found
#16 159.0 + hide_output make -j2
#16 159.0 + set +x
#16 159.0 + set +x
#16 159.0 trap: ERR: bad trap
#16 ERROR: process "/bin/sh -c sh /scripts/build-gccjit.sh /scripts" did not complete successfully: exit code: 1
 > [11/11] RUN sh /scripts/build-gccjit.sh /scripts:
157.6 + pwd
157.6 + ../gcc-src/configure --enable-host-shared --enable-languages=jit --enable-checking=release --disable-bootstrap --disable-multilib --prefix=/scripts/gcc-build/../gcc-install --quiet
158.2 The following languages will be built: c,jit,lto
158.2 The following languages will be built: c,jit,lto
158.6 *** This configuration is not supported in the following subdirectories:
158.6      gnattools gotools libgrust target-libada target-libstdc++-v3 target-libphobos target-zlib target-libbacktrace target-libgfortran target-libgo target-libffi target-libgm2 target-libobjc target-libgrust target-libitm target-libsanitizer target-libvtv
158.6 /scripts/gcc-src/missing: 81: makeinfo: not found
159.0 + hide_output make -j2
159.0 + set +x
159.0 + set +x
159.0 trap: ERR: bad trap
Dockerfile:101
--------------------
  99 |     COPY host-x86_64/dist-x86_64-linux/build-gccjit.sh /scripts/
 100 |     
```

</details>

Mark-Simulacrum commented:
https://github.com/rust-lang/rust/blob/master/src/ci/docker/host-x86_64/dist-x86_64-linux/build-clang.sh#L42 is an example. It also demonstrates the use of hide_output.

rust-log-analyzer commented:
The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8296214682/job/22704809623) [(plain)](https://github.com/rust-lang/rust/commit/0a1b64bed781e6ac51fc61a65eaeb9cfb821f030/checks/22704809623/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:b4ffde65f46336ab88eb53be808477a3936bae11)
Download action repository 'actions/upload-artifact@v3' (SHA:a8a3f3ad30e3422c9c7b888a15615d19a852ae32)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#10 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#10 DONE 0.0s

#11 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#11 0.567   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#11 0.580 Collecting boolean-py==4.0
#11 0.583   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#11 0.594 Collecting chardet==5.1.0
---
#11 3.576 Building wheels for collected packages: reuse
#11 3.577   Building wheel for reuse (pyproject.toml): started
#11 3.904   Building wheel for reuse (pyproject.toml): finished with status 'done'
#11 3.904   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#11 3.905   Stored in directory: /tmp/pip-ephem-wheel-cache-qxex1r3k/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#11 3.907 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#11 3.929   Attempting uninstall: setuptools
#11 3.930     Found existing installation: setuptools 59.6.0
#11 3.931     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#11 3.931     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#11 3.932     Can't uninstall 'setuptools'. No files were found to uninstall.
#11 4.608 Successfully installed binaryornot-0.4.4 boolean-py-4.0 chardet-5.1.0 jinja2-3.1.2 license-expression-30.0.0 markupsafe-2.1.1 python-debian-0.1.49 reuse-1.1.0 setuptools-66.0.0
#11 4.609 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#11 5.117 Collecting virtualenv
#11 5.158   Downloading virtualenv-20.25.1-py3-none-any.whl (3.8 MB)
#11 5.214      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.8/3.8 MB 70.1 MB/s eta 0:00:00
#11 5.250 Collecting distlib<1,>=0.3.7
#11 5.255   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#11 5.289 Collecting filelock<4,>=3.12.2
#11 5.292   Downloading filelock-3.13.1-py3-none-any.whl (11 kB)
#11 5.292   Downloading filelock-3.13.1-py3-none-any.whl (11 kB)
#11 5.321 Collecting platformdirs<5,>=3.9.1
#11 5.324   Downloading platformdirs-4.2.0-py3-none-any.whl (17 kB)
#11 5.409 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#11 5.574 Successfully installed distlib-0.3.8 filelock-3.13.1 platformdirs-4.2.0 virtualenv-20.25.1
#11 DONE 5.7s

#12 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#12 DONE 0.0s
---
DirectMap4k:      192448 kB
DirectMap2M:     6098944 kB
DirectMap1G:    12582912 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
    Finished dev [unoptimized] target(s) in 0.03s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/d7723b21910075a3c38c2b6e64b2467394c93724/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-d7723b21910075a3c38c2b6e64b2467394c93724-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
##[endgroup]
fmt check
tidy check
tidy: Skipping binary file check, read-only filesystem
##[error]tidy error: /checkout/src/ci/docker/host-x86_64/dist-x86_64-linux/build-gccjit.sh:1: line longer than 100 chars
##[error]tidy error: /checkout/src/ci/docker/host-x86_64/dist-x86_64-linux/build-gccjit.sh:1: trailing whitespace
removing old virtual environment
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10'
Requirement already satisfied: pip in ./build/venv/lib/python3.10/site-packages (24.0)
Collecting black==23.3.0 (from -r /checkout/src/tools/tidy/config/requirements.txt (line 7))
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.7/1.7 MB 37.7 MB/s eta 0:00:00
Collecting click==8.1.3 (from -r /checkout/src/tools/tidy/config/requirements.txt (line 34))
  Downloading click-8.1.3-py3-none-any.whl (96 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 96.6/96.6 kB 24.5 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 96.6/96.6 kB 24.5 MB/s eta 0:00:00
Collecting importlib-metadata==6.7.0 (from -r /checkout/src/tools/tidy/config/requirements.txt (line 38))
  Downloading importlib_metadata-6.7.0-py3-none-any.whl (22 kB)
Collecting mypy-extensions==1.0.0 (from -r /checkout/src/tools/tidy/config/requirements.txt (line 42))
  Downloading mypy_extensions-1.0.0-py3-none-any.whl (4.7 kB)
Collecting packaging==23.1 (from -r /checkout/src/tools/tidy/config/requirements.txt (line 46))
  Downloading packaging-23.1-py3-none-any.whl (48 kB)
Collecting pathspec==0.11.1 (from -r /checkout/src/tools/tidy/config/requirements.txt (line 50))
  Downloading pathspec-0.11.1-py3-none-any.whl (29 kB)
Collecting platformdirs==3.6.0 (from -r /checkout/src/tools/tidy/config/requirements.txt (line 54))
  Downloading platformdirs-3.6.0-py3-none-any.whl (16 kB)
  Downloading platformdirs-3.6.0-py3-none-any.whl (16 kB)
Collecting ruff==0.0.272 (from -r /checkout/src/tools/tidy/config/requirements.txt (line 58))
  Downloading ruff-0.0.272-py3-none-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (5.9 MB)
Collecting tomli==2.0.1 (from -r /checkout/src/tools/tidy/config/requirements.txt (line 77))
  Downloading tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting typed-ast==1.5.4 (from -r /checkout/src/tools/tidy/config/requirements.txt (line 81))
  Downloading typed_ast-1.5.4-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (877 kB)
  Downloading typed_ast-1.5.4-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (877 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 877.7/877.7 kB 84.7 MB/s eta 0:00:00
Collecting typing-extensions==4.6.3 (from -r /checkout/src/tools/tidy/config/requirements.txt (line 107))
  Downloading typing_extensions-4.6.3-py3-none-any.whl (31 kB)
Collecting zipp==3.15.0 (from -r /checkout/src/tools/tidy/config/requirements.txt (line 114))
  Downloading zipp-3.15.0-py3-none-any.whl (6.8 kB)
Installing collected packages: zipp, typing-extensions, typed-ast, tomli, ruff, platformdirs, pathspec, packaging, mypy-extensions, click, importlib-metadata, black
Successfully installed black-23.3.0 click-8.1.3 importlib-metadata-6.7.0 mypy-extensions-1.0.0 packaging-23.1 pathspec-0.11.1 platformdirs-3.6.0 ruff-0.0.272 tomli-2.0.1 typed-ast-1.5.4 typing-extensions-4.6.3 zipp-3.15.0
some tidy checks failed
Build completed unsuccessfully in 0:01:02
  local time: Fri Mar 15 12:36:11 UTC 2024
  network time: Fri, 15 Mar 2024 12:36:11 GMT
```

</details>

RalfJung commented on src/ci/docker/host-x86_64/dist-x86_64-linux/build-gccjit.sh:
```diff
@@ -14,15 +16,17 @@ mkdir ../gcc-build ../gcc-install
-make install
+    --prefix=$(pwd)/../gcc-install \
+
+hide_output make -j$(nproc)
```
This seems to speed up the build by a factor of 4.^^ It was 30min before, now it's...
```
2024-03-15T12:47:03.8776557Z #14 DONE 446.8s
```

GuillaumeGomez replied:
Nice. :)

RalfJung commented on src/ci/docker/host-x86_64/dist-x86_64-linux/build-gccjit.sh:
```diff
@@ -1,7 +1,9 @@
 
 set -ex
 
+source /tmp/shared.sh
```
Why didn't you put this in the same folder as the build-gccjit script?

GuillaumeGomez replied:
Mostly because I was first trying to make it work. Cleanup time!

rust-log-analyzer commented:
A job failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8298281635/job/22711312805) [(plain)](https://github.com/rust-lang/rust/commit/86a0c67d3313ef6dd7724298fb064f87e3e1ed5b/checks/22711312805/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

```

</details>

GuillaumeGomez commented:
Clean up done. Nice speed up in any case. :)

Mark-Simulacrum commented:
@bors r+
